modules/cluster/main.tf

```diff
@@ -0,0 +1,160 @@
+# // GKE Cluster
+data "google_compute_network" "default" {
+  name = "default"
+}
+
+data "google_compute_subnetwork" "default" {
+  name   = "default"
+  region = var.gcp_region
+}
+
+data "google_client_config" "default" {}
+
+data "google_artifact_registry_docker_image" "my_image" {
+  location      = var.gcp_region
+  repository_id = var.google_artifact_registry_repository_id
+  image_name    = "agent-image"
+}
```

Comment:
This image doesn't exist, because the cloud build pipeline hasn't been triggered to push the docker image to the artifact registry.